dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java

```diff
@@ -532,6 +534,22 @@ public Workflow getWorkflow(@ApiParam(hidden = true) @Parameter(hidden = true, n
         return workflow;
     }
 
+    @GET
+    @Path("/{workflowId}/workflowVersions")
+    @UnitOfWork
+    @ApiOperation(nickname = "getWorkflowVersions", value = "Return all versions in an entry", authorizations = {
+            @Authorization(value = JWT_SECURITY_DEFINITION_NAME) }, response = WorkflowVersion.class, responseContainer = "List", notes = "This is one of the few endpoints that returns the user object with populated properties (minus the userProfiles property)")
+    @Operation(operationId = "getWorkflowVersions", description = "Return all versions in an entry", security = @SecurityRequirement(name = OPENAPI_JWT_SECURITY_DEFINITION_NAME))
+    public Set<WorkflowVersion> getWorkflowVersions(@ApiParam(hidden = true) @Parameter(hidden = true, name = "user")@Auth User user,
```

Comment:
Didn't Gary write some code that returns n versions to the UI? How does this interact with that?

Reply:
> Not knowing this code as well as I should, and not seeing it off-hand when skimming, what happens if a version is deleted? Do the entry-level fileformats get updated?

I added code to update the entry-level fileformats in the same function that updates all the version level ones. But it looks like that only gets called on refresh (for tools and workflows) and updating hosted entries. So this should handle deleted versions when refreshing, but not for deleting a hosted version. And it looks like we don't add file formats at the version level for .dockstore.yml workflows. I'll add this.

Reply:
> And it looks like we don't add file formats at the version level for .dockstore.yml workflows... But doing so would load all the versions at then defeating this PR?

Hmmm, shouldn't it be possible to add file formats for the version being updated without updating the entry-level fileformats? Maybe we need to update the entry-level .dockstore.yml fileformats in batches or on a schedule?

(definitely a different ticket, but good to know)

Reply:
Yeah, so I added some code that will update the file formats for the version that is being added/updated and then it updates the entry by just trying to add those to the entry's set. But it doesn't try to update all versions we already have for the workflow.

Reply:
> Didn't Gary write some code that returns n versions to the UI? How does this interact with that?

Think about this further, should this endpoint have a query parameter with a maximum number of versions, with a default of 25 or whatever @garyluu came up with?

And ultimately I wonder it should support pagination.

Reply:
Just added what Gary came up with. The limit is 200.